src/components/breadcrumb-bar/utils/get-docs-breadcrumbs.ts

```diff
@@ -63,17 +63,22 @@ function getPathMatchedNode(navNodes, pathString) {
   if (matches.length > 1) {
     if (IS_DEV) {
       console.warn(
-        `Ambiguous breadcrumb path: Found ${matches.length} matches for "${pathString}". Please ensure there is exactly one node or index-less category with the path "${pathString}" in the provided navData.`
+        `Ambiguous breadcrumb path under "${basePath}": Found ${matches.length} matches for "${pathString}". Please ensure there is exactly one node or index-less category with the path "${pathString}" in the provided navData.`
       )
     }
     return matches[0]
   }
   // Otherwise, we have zero matches, which would mean a breadcrumb with missing parts.
-  // This would feel broken to a site visitor, so should be a blocking
-  // issue we need to fix, so we throw an error.
-  throw new Error(
```

Comment:
Previously we threw an error here. But, this broke the build when rendering [`/nomad/docs/autoscaler/agent/source`](https://dev-portal-git-zsadd-nomad-docs-views-hashicorp.vercel.app/nomad/docs/autoscaling/agent/source) due to some quirks in Nomad's nav-data (described in more detail in the PR description).

To fix this, we now render a kind of "fallback" breadcrumb bar item if we can't match up a nav node. This item is:
1. Unlinked, because there's nothing to link to
2. Titled with the raw `path` part to the page, because there's no nav-node with a pretty title.

We definitely want to avoid these situations, so I've retained warnings here. We may also be able to more strictly validate `nav-data` in order to prevent these types of nav-data structures at the source.